airflow/models.py

```diff
@@ -1022,12 +1022,25 @@ def are_dependents_done(self, session=None):
     @provide_session
     def previous_ti(self, session=None):
         """ The task instance for the task that ran before this task instance """
-        return session.query(TaskInstance).filter(
-            TaskInstance.dag_id == self.dag_id,
-            TaskInstance.task_id == self.task.task_id,
-            TaskInstance.execution_date ==
-            self.task.dag.previous_schedule(self.execution_date),
-        ).first()
+        dag = self.task.dag
+
+        if dag.backfill:
+            return session.query(TaskInstance).filter(
+                TaskInstance.dag_id == self.dag_id,
+                TaskInstance.task_id == self.task.task_id,
+                TaskInstance.execution_date ==
+                self.task.dag.previous_schedule(self.execution_date),
```

Comment:
I know, but as you are changing the function anyway it would be nice to have it fixed properly.